### PR TITLE
Add role/cluster binding for nfs-provisioner

### PR DESCRIPTION
**Pull request by khushboo-rancher**
https://github.com/longhorn/longhorn/issues/1606

I'm not sure about ```service``` ```longhorn-nfs-provisioner``` that it should be included here or not.

Signed-off-by: khushboo-rancher <60111667+khushboo-rancher@users.noreply.github.com>

### CODE REVIEW

**Comment by shuo-wu** [comment]
The RBAC is already in `examples/rwx`. Maybe we can put the enhanced PSP in that directory if the existing PSP is editable.

**Comment by khushboo-rancher** [comment]
As per my understanding, going forward to address this https://github.com/longhorn/longhorn/issues/1470, we can modify the existing podsecuritypolicy.yaml and add serviceAccount ```longhorn-nfs-provisioner```.

@joshimoo Any thoughts?

File: deploy/install/01-prerequisite/02-serviceaccount.yaml
Comment (joshimoo) [inline deploy/install/01-prerequisite/02-serviceaccount.yaml:69]: Service does not belong here.

**Review by joshimoo — CHANGES_REQUESTED** [review]
The service account change looks good, but since we anyway give the PSP the permissions.
We might just want to only use a single service-account @yasker ?

From a security perspective it's always better to have lowest common security profiles, but we don't really do that at the moment, so it might be okay to just have a single account for now.

**Comment by yasker** [comment]
I am fine with using the same service account.

And the service part should be in the NFS provisioner deployment, at least before we have the HA RWX.

**Comment by yasker** [comment]
Also, need to update the sign-off-by.

**Comment by yasker** [comment]
Since we're merging the PSP/role binding from `examples/rwx`, we should also update the `examples/rwx` to use the Longhorn service account, etc. Otherwise, this change has no users.

**Comment by khushboo-rancher** [comment]
Discussed with @yasker, we need to change the ```example/rwx``` and doc of rwx too. We can wait till the story ```HA rwx``` is delivered and then merge this to have all the relevant changes at one time.

**Comment by joshimoo** [comment]
Closing this, since the changes are no longer necessary.
Once the native rwx support is merged, https://github.com/longhorn/longhorn-manager/pull/778
We can then remove the rwx provisioner example, and transition to using the native rwx support.

**Pull request closed without merging**
